### PR TITLE
Actually apply block and dodge bonuses

### PR DESCRIPTION
**Pull request by paroid01**
#### Summary
SUMMARY: Bugfixes "Applies bonus blocks and dodges from martial arts and removed block for being grabbed"

#### Purpose of change
In testing, I discovered that a refactoring at some point had prevented
the application of increased or decreased block and dodge count.

#### Describe the solution
Changes the martial arts buffs that modify block and dodge to apply to
the character's num blocks bonus. The player's blocks and dodges left is
initialized to the blocks and dodges plus the bonus during the start of
the player's turn.

#### Describe alternatives you've considered
Reordering the application of buffs versus the initialization of the block and
dodges left variables could have fixed the problem, but would probably have
lots of unintended consequences.

#### Additional context
Being grabbed by a zombie will actually take your base block per turn away
now.

### CODE REVIEW

File: src/martialarts.cpp [hunk 1]
```diff
@@ -471,8 +471,8 @@ bool ma_buff::is_valid_player( const player &u ) const
 
 void ma_buff::apply_player( player &u ) const
 {
-    u.dodges_left += dodges_bonus;
-    u.blocks_left += blocks_bonus;
+    u.set_num_dodges_bonus(u.get_num_dodges_bonus() + dodges_bonus);
```
Comment (ZhilkinSerg) [inline src/martialarts.cpp]:
```suggestion
    u.set_num_dodges_bonus( u.get_num_dodges_bonus() + dodges_bonus );
```

File: src/martialarts.cpp [hunk 2]
```diff
@@ -471,8 +471,8 @@ bool ma_buff::is_valid_player( const player &u ) const
...
+    u.set_num_blocks_bonus(u.get_num_blocks_bonus() + blocks_bonus);
```
Comment (ZhilkinSerg) [inline src/martialarts.cpp]:
```suggestion
    u.set_num_blocks_bonus( u.get_num_blocks_bonus() + blocks_bonus );
```

File: src/player_hardcoded_effects.cpp [hunk 3]
```diff
@@ -840,7 +840,7 @@ void player::hardcoded_effects( effect &it )
             hp_cur[hp_torso] = 0;
         }
     } else if( id == effect_grabbed ) {
-        blocks_left -= 1;
+        set_num_blocks_bonus(get_num_blocks_bonus() - 1);
```
Comment (ZhilkinSerg) [inline src/player_hardcoded_effects.cpp]:
```suggestion
        set_num_blocks_bonus( get_num_blocks_bonus() - 1 );
```

File: src/martialarts.cpp (hunk 1 above)
Comment (paroid01) [reply src/martialarts.cpp]: I need to figure out how to put astyle into my visual studio, don't I.

Comment (ZhilkinSerg) [reply src/martialarts.cpp]: See https://github.com/CleverRaven/Cataclysm-DDA/pull/24006 for instruction.

Comment (paroid01) [reply src/martialarts.cpp]: Thanks! I think I've gotten that working.

**Pull request merged**